snabbdom/src/test/scala/com/github/buntec/snabbdom/AttributesSuite.scala

```diff
@@ -0,0 +1,206 @@
+            .build
+        )
+        val elm = patch(vnode0, vnode1).elm.get.asInstanceOf[dom.Element]
+        assertEquals(elm.hasAttribute("href"), true)
+        assertEquals(elm.hasAttribute("minlength"), true)
+        assertEquals(elm.hasAttribute("value"), true)
+        assertEquals(elm.hasAttribute("title"), true)
+    }
+
+    vnode0.test("are set correctly when namespaced") { vnode0 =>
+      val vnode1 =
+        h("div", VNodeData.builder.withAttrs("xlink:href" -> "#foo").build)
+      val elm = patch(vnode0, vnode1).elm.get.asInstanceOf[dom.Element]
+      assertEquals(
+        elm.getAttributeNS("http://www.w3.org/1999/xlink", "href"),
+        "#foo"
+      )
+    }
+
+    test("should not touch class nor id fields") {
+      val elm = dom.document.createElement("div").asInstanceOf[dom.HTMLElement]
+      elm.id = "myId"
+      elm.className = "myClass"
+      val vnode0 = elm
+      val vnode1 = h(
+        "div#myId.myClass",
+        VNodeData.builder.withAttrs().build,
+        Array[VNode]("Hello")
+      )
+      val elm1 = patch(vnode0, vnode1).elm.get.asInstanceOf[dom.HTMLElement]
+      assertEquals(elm1.tagName, "DIV")
+      assertEquals(elm1.id, "myId")
+      assertEquals(elm1.className, "myClass")
+      assertEquals(elm1.textContent, "Hello")
+    }
+
+  }
+
+  group("boolean attribute") {
+
+    vnode0.test("is present and empty string if the value is thruthy (in JS)") {
+      vnode0 =>
+        val vnode1 = h(
+          "div",
+          VNodeData.builder
+            .withAttrs(
+              "required" -> true,
+              "readonly" -> 1,
+              "noresize" -> "truthy"
+            )
+            .build
+        )
+        val elm = patch(vnode0, vnode1).elm.get.asInstanceOf[dom.HTMLElement]
+        assertEquals(elm.hasAttribute("required"), true)
+        assertEquals(elm.getAttribute("required"), "")
+        assertEquals(elm.hasAttribute("readonly"), true)
+        assertEquals(elm.getAttribute("readonly"), "1")
+        assertEquals(elm.hasAttribute("noresize"), true)
+        assertEquals(elm.getAttribute("noresize"), "truthy")
+    }
+
+    vnode0.test("is omitted if the value is false") { vnode0 =>
+      val vnode1 =
+        h("div", VNodeData.builder.withAttrs("required" -> false).build)
+      val elm = patch(vnode0, vnode1).elm.get.asInstanceOf[dom.HTMLElement]
+      assertEquals(elm.hasAttribute("required"), false)
+      assertEquals(elm.getAttribute("required"), null)
+    }
+
+    vnode0.test("is not omitted if the value is falsy") { vnode0 =>
+      val vnode1 =
+        h(
+          "div",
+          VNodeData.builder.withAttrs("readonly" -> 0, "noresize" -> "").build
+        )
+      val elm = patch(vnode0, vnode1).elm.get.asInstanceOf[dom.HTMLElement]
+      assertEquals(elm.hasAttribute("readonly"), true)
+      assertEquals(elm.hasAttribute("noresize"), true)
+    }
+
+  }
+
+  group("Object.prototype property") {
+
+    // not sure I understand this test
```

Comment:
Some discussion here:
- https://github.com/snabbdom/snabbdom/pull/161

I think if we only use `String` for attributes it won't apply.